src/embed/http.rs

```diff
@@ -0,0 +1,91 @@
+use std::convert::TryFrom;
+use std::error::Error;
+use std::fmt;
+use std::str;
+use wasm_bindgen::JsCast;
+use wasm_bindgen::JsValue;
+use wasm_bindgen_futures::JsFuture;
+use web_sys::{RequestInit, RequestMode};
+
+// browser_fetch makes and HTTP request over the network via the browser's Fetch API.
+// It is intended to be used in wasm; it won't work in regular rust. The response contains
```

Comment:
[idle speculation - no task for now] It probably makes more sense to turn `src/wasm.rs` into `src/wasm` and put this there along with some abstraction from the "core" to the correct version.